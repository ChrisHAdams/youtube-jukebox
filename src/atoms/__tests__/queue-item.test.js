import React from "react"
import renderer from "react-test-renderer";
import QueueItem from "../queue-item";

import {render, fireEvent, cleanup} from '@testing-library/react'
import {toHaveStyle} from '@testing-library/jest-dom';

expect.extend({toHaveStyle})

const videoMock = {
  id: "someId",
  title: "Some Title",
  thumbnails:{
    default:{
      url:"www.something"
    }
  }

}


describe("QueueItem", () => {

  afterEach(cleanup);

  it("renders correctly", () => {
    const tree = renderer
      .create(<QueueItem
                key="keyVal"
                video={videoMock}
                lightTheme={false}
                moveVideoUpQueue="callback"
                moveVideoDownQueue="callback"
                removeFromQueue="callback"
                playNow="false"/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls "onClick" prop on button click', () => {
    // Render new instance in every test to prevent leaking state
    const upOnClick = jest.fn();
    const downOnClick = jest.fn();
    const removeOnClick = jest.fn();
    const playNowOnClick = jest.fn();
    const lightThemeValue = false;

    const { getByTestId } = render(<QueueItem
                                  key="keyVal"
                                  video={videoMock}
                                  lightTheme={lightThemeValue}
                                  moveVideoUpQueue={upOnClick}
                                  moveVideoDownQueue={downOnClick}
                                  removeFromQueue={removeOnClick}
                                  playNow={playNowOnClick}/>);

    fireEvent.click(getByTestId('up'));
    expect(upOnClick).toHaveBeenCalled();

    fireEvent.click(getByTestId('down'));
    expect(downOnClick).toHaveBeenCalled();

    fireEvent.click(getByTestId('remove'));
    expect(removeOnClick).toHaveBeenCalled();

    fireEvent.click(getByTestId('playNow'));
    expect(playNowOnClick).toHaveBeenCalled();

    expect(getByTestId('container')).toHaveStyle(`background-color: '#293447'`);

  });

  it('calls "onClick" prop on button click', () => {
    // Render new instance in every test to prevent leaking state
    const upOnClick = jest.fn();
    const downOnClick = jest.fn();
    const removeOnClick = jest.fn();
    const playNowOnClick = jest.fn();
    const lightThemeValue = true;

    const { getByTestId } = render(<QueueItem
                                  key="keyVal"
                                  video={videoMock}
                                  lightTheme={lightThemeValue}
                                  moveVideoUpQueue={upOnClick}
                                  moveVideoDownQueue={downOnClick}
                                  removeFromQueue={removeOnClick}
                                  playNow={playNowOnClick}/>);


    expect(getByTestId('container')).toHaveStyle(`background-color:'#293447'`);

  });

})


