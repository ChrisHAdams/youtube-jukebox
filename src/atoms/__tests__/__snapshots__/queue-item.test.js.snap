// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueueItem renders correctly 1`] = `
<div>
  <div
    className="sc-bwzfXH bhysCS"
    data-testid="container"
  >
    <img
      alt="Some Title"
      className="sc-htpNat cPBnlB"
      data-testid="playNow"
      onClick={[Function]}
      src="www.something"
      title="Some Title"
    />
    <p
      className="sc-EHOje fdXWyM"
    >
      Some Title
    </p>
    <div
      className="sc-bxivhb EYjjH"
    >
      <div
        className="sc-ifAKCX jAAoci"
      >
        <svg
          data-testid="up"
          fill="currentColor"
          height="2em"
          onClick={[Function]}
          size="2em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="2em"
        >
          <path
            d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm292 116V256h70.9c10.7 0 16.1-13 8.5-20.5L264.5 121.2c-4.7-4.7-12.2-4.7-16.9 0l-115 114.3c-7.6 7.6-2.2 20.5 8.5 20.5H212v116c0 6.6 5.4 12 12 12h64c6.6 0 12-5.4 12-12z"
          />
        </svg>
      </div>
      <div
        className="sc-ifAKCX jAAoci"
      >
        <svg
          data-testid="down"
          fill="currentColor"
          height="2em"
          onClick={[Function]}
          size="2em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="2em"
        >
          <path
            d="M504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zM212 140v116h-70.9c-10.7 0-16.1 13-8.5 20.5l114.9 114.3c4.7 4.7 12.2 4.7 16.9 0l114.9-114.3c7.6-7.6 2.2-20.5-8.5-20.5H300V140c0-6.6-5.4-12-12-12h-64c-6.6 0-12 5.4-12 12z"
          />
        </svg>
      </div>
      <div
        className="sc-ifAKCX jAAoci"
      >
        <svg
          data-testid="remove"
          fill="currentColor"
          height="2em"
          onClick={[Function]}
          size="2em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="2em"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
